includes/Modules/Analytics/Google_Service_AnalyticsProvisioning.php

```diff
@@ -0,0 +1,49 @@
+<?php
+/**
+ * Class Google\Site_Kit\Modules\Analytics\Google_Service_AnalyticsProvisioning
+ *
+ * @package   Google\Site_Kit\Modules\Analytics
+ * @copyright 2020 Google LLC
+ * @license   https://www.apache.org/licenses/LICENSE-2.0 Apache License 2.0
+ * @link      https://sitekit.withgoogle.com
+ */
+
+namespace Google\Site_Kit\Modules\Analytics;
+
+use Google\Site_Kit_Dependencies\Google_Service_Analytics;
+use Google\Site_Kit_Dependencies\Google_Client;
+
+/**
+ * Class for Analytics Provisioning service.
+ *
+ * @since n.e.x.t
+ * @access private
+ * @ignore
+ */
+class Google_Service_AnalyticsProvisioning extends Google_Service_Analytics {
+	/**
+	 * Constructs the internal representation of the Analytics service.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param Google_Client $client The client used to deliver requests.
+	 * @param string        $rootUrl The root URL used for requests to the service.
+	 */
+	public function __construct( Google_Client $client, $rootUrl = null ) { // phpcs:ignore WordPress.NamingConventions.ValidVariableName
+		parent::__construct( $client );
```

Comment:
I don't think this would have worked as I had it. `Google_Service_AnalyticsProvisioning` was a recent addition so I may not have tested the entire flow. Without this URL I would expect the client to query the service directly and have the redirectUri rejected.